### PR TITLE
[Merged by Bors] - chore: adaptations for nightly-2024-03-24

### PR DESCRIPTION
**Pull request by kim-em**
There are some regressions in `rw?` and `rw_search`, so some test cases are commented out. Joe is looking at these today, and they are not used in committed code in Mathlib, so I think it is okay to proceed as is.

### CODE REVIEW

File: Archive/Examples/IfNormalization/WithoutAesop.lean [hunk 1]
```diff
@@ -66,27 +66,27 @@ def normalize' (l : AList (fun _ : ℕ => Bool)) :
             refine ⟨fun _ => ?_, fun _ => ?_⟩
             · congr
               ext w
-              by_cases h : w = v <;> rename_i x
+              by_cases h : w = v
```
Comment (semorrison) [inline Archive/Examples/IfNormalization/WithoutAesop.lean:69]: New unused variables linter.

File: .github/workflows/bors.yml [hunk 2]
```diff
@@ -79,10 +79,6 @@ jobs:
         run: |
           git ls-files 'Mathlib/*.lean' | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib.lean
 
-      - name: update MathlibExtras.lean
-        run: |
-          git ls-files 'MathlibExtras/*.lean' | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > MathlibExtras.lean
-
```
Comment (semorrison) [inline .github/workflows/bors.yml:85]: With new `rw?` implementation, we can throw out the `MathlibExtras` machinery! No more caches.

**Review by kmill — APPROVED** [review]
Looks good to me. Just approving in case anyone else wants to take a look at it, but I'll merge in the next few hours (assuming someone already hasn't).

**Pull request closed without merging**